### DISCUSSION TITLE
Library Organizer: Different year formats for completed/uncompleted/short series?

### DISCUSSION THREAD

**Discussion by Cantello** [maforget/ComicRackCE#39] [Q&A] [answered]
Hi all!

I'd like to organise my (small) library a bit better and intended to add the running years for each series but as I have many Graphic Novels and mini-series, I would like to have three different date formats:

- for a completed, longer-running series: (StartYear-EndYear) 
- for a completed mini-series/GN: (StartYear)
- for an ongoing longer-running series: (StartYear-)  (to differentiate between this and a mini series)

Is this possible? I currently have this:
`\{ <language>}\{ <seriesgroup>}\{<series>}{ - V<volume0>} { ({<startyear>}<!startyear({<EndYear>})>-{<EndYear>})}{ ({<startyear>}<?startyear({<EndYear>})>)}`

but I cannot figure out how to have the trailing dash insererted depending on the completeness status.
Any ideas?

**Comment by maforget** [comment, marked as answer]
~~Something like this will give you (StartYear-):~~
~~`{<language>}\{ <seriesgroup>}\{<series>}{ - V<volume0>} { ({<startyear>}<!startyear({<EndYear>})>-{<EndYear>})}{ ({<startyear>}<?startyear({<EndYear>})>{<seriesComplete(- )(!)>)}}`~~

~~I just used the UI to insert this into you already done rule. You need to have something in the text field (in this case "- "), but you can't have a parenthesis, that you need to put it in the suffix.~~

Edit: This is wrong is is your magic line:
`{({<startyear>}<!startyear({<EndYear>})>{<seriesComplete(-)(!)>}{<seriesComplete({-<EndYear>})>})}`

**Pseudo Code:**
```python
if startYear != EndYear:
     prefix = "(" + startYear
     if seriesComplete:
         suffix = "-"
     else:
         suffix = "-" + EndYear
     suffix = suffix + ")"
```  

So your complete code is:
`{ <language>}\{ <seriesgroup>}\{ <series>}{ - V<volume0>} { ({<startyear>}<!startyear({<EndYear>})>{<seriesComplete(-)(!)>}{<seriesComplete({-<EndYear>})>})}{ ({<startyear>}<?startyear({<EndYear>})>)}`

**Reply by Cantello** [reply]
I tried it and it works! Easier than using the rules but I am glad I found out about how rules work anyway. Thanks!
